### ISSUE TITLE
Template arguments don't work with custom shaders

### ISSUE THREAD

**Issue by phr00t** [phr00t/FocusEngine#90]
https://doc.xenko.com/latest/en/manual/graphics/effects-and-shaders/custom-shaders.html#arguments-and-parameters

Trying the above just seems to break shaders. I do notice the custom material template project shader works when first loaded, but if you try and modify the template arguments -- the "Generics" list in the editor doesn't get updated and the shader fails to compile. Adding new template arguments don't seem to add themselves to "Generics". Trying to use template arguments in other shaders just causes them to crash and the editor to crash (I made the editor skip rendering broken meshes from this as a workaround).

Basically, you can't use template arguments reliably. The workaround is to use parameters (which have been tested to work) or Compositions (not tested by myself personally).

**Comment by phr00t** [comment]
Compositions don't seem to work either. Parameters seem to be the only flexible & reliable method of getting values to custom shaders.

**Comment by phr00t** [comment]
Removing the shader with template arguments (clearing out the field) and reselecting it for the material seems to refresh the template arguments in the editor. When it gets refreshed, it seems to work!
